src/python/grapl_analyzerlib/grapl_analyzerlib/queryable.py

```diff
@@ -0,0 +1,197 @@
+import abc
+import json
+import functools
+
+from grapl_analyzerlib.comparators import _str_cmps, _int_cmps
+from collections import defaultdict
+
+from typing import (
+    cast,
+    Any,
+    TypeVar,
+    Generic,
+    Type,
+    Tuple,
+    Optional,
+    Iterator,
+    List,
+    Union,
+)
+
+from uuid import uuid4
+
+from grapl_analyzerlib.extendable import Extendable
+
+Q = TypeVar("Q", bound="Queryable")
+V = TypeVar("V", bound="Viewable")
+
+F = TypeVar("F", bound="Queryable")
+
+ToOneFilter = List[F]
+ToManyFilter = List[Tuple[F, ...]]
+EdgeFilter = Union[ToOneFilter[F], ToManyFilter[F]]
+
+
+def with_str_prop(prop):
+    @functools.wraps(prop)
```

Comment:
wait, okay, I'm confused. `prop` here is just a string name of the property. What the heck does @functools.wraps(some_string) do?